### ISSUE TITLE
Add --image-type flag to upgrade.sh

### ISSUE THREAD

**Issue by davidopp** [kubernetes/kubernetes#32552]
To select GCI vs. ContainerVM

Also, if it will change the image type, we should warn the user that the upgrade will change the image, and prompt them whether to continue [yes/no] (suggestion by @dchen1107)

@roberthbailey @fabioy @dchen1107 @vishh

**Comment by dchen1107** [comment]
I prefer not adding --image-type flag to upgrade-related scripts to do this. My concern is that the choice is not persistent through this way.

But I think warning the user that the upgrade will change the image, and prompt them whether to continue is good. If they want to opt-out gci image, the script prints out how to change the default choice at: https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/config-default.sh#L39

**Comment by vishh** [comment]
Would setting env `KUBE_NODE_OS_DISTRIBUTION` to either `gci` or `debian`
suffice this use case? This is a transitional phase and this option will
not be relevant in the future.

**Comment by vishh** [comment]
The upgrade scripts are preserving the existing base image. I wonder if they ever upgraded container-vm based on the version pinned to a release.
This is an issue because users will be stuck with old versions of the base image as long as they continue to use this script.

**Comment by vishh** [comment]
I talked to @dchen1107 offline. Here is the summary for v1.4.0 release.

1. upgrade.sh script does not handle base image upgrades. This is an issue by itself. But it is not a blocker for v1.4.0 because the base image has not been upgraded since v1.3.0.
2. upgrade.sh does not support upgrading to gci by default. This is a feature by itself at-least for v1.4.0

After v1.4.0 frenzy is over, we need to fix `1` and hopefully include the fix in patch releases of v1.4. `2` needs to be fixed before v1.5.

**Comment by vishh** [comment]
@davidopp I'd recommend not including this issue as a v1.4.0 release blocker.

**Comment by davidopp** [comment]
@vishh Can you take ownership of both of the items?

**Comment by vishh** [comment]
If @mtaufen cannot get to it sometime soon (< week), I will pick this up.

On Wed, Sep 14, 2016 at 9:51 PM, David Oppenheimer <notifications@github.com

**Comment by davidopp** [comment]
According to #32812 upgrade.sh actually does install GCI when you go from 1.3 to 1.4 (does it install ContainerVM when you go the other direction?) So I think that means this issue is now a release blocker.

**Comment by vishh** [comment]
I think I used upgrade scripts from v1.3.x to upgrade to v1.4.0. Bummer.
Working on a fix for this with Lantao.

On Thu, Sep 15, 2016 at 4:31 PM, David Oppenheimer <notifications@github.com

**Comment by dchen1107** [comment]
@davidopp This is not P0, and should't block 1.4 release. The decision we agreed upon is that upgrade.sh shouldn't upgrade the underneath os image for 1.4 release. #32812 is p0 blocker for 1.4, and @Random-Liu is working on it actively.

**Comment by davidopp** [comment]
I agree, fixing this by just not having upgrade.sh update the OS image is a good fix for 1.4. Sorry for the confusion.

**Comment by Random-Liu** [comment]
The fix is here https://github.com/kubernetes/kubernetes/pull/32840.
With #32840, we can also specify image type with flag `-o`:
```
-o:  Use os distro sepcified in KUBE_NODE_OS_DISTRIBUTION for new nodes
```

**Comment by dims** [comment]
This needs to be triaged as a release-blocker or not for 1.5 @davidopp @dchen1107 

From previous comment this seems to have been fixed already, so @Random-Liu please close this if appropriate

**Comment by dchen1107** [comment]
This is not a release blocker, and the urgent fix for the initial blocking problem was addressed before 1.4. 

EDIT: I decided to close the issue based on my above comment. The work of improving upgrade mechanism for image option is handled by mtaufen@ separately.
